backend/forecastbox/api/routers/job.py

```diff
@@ -140,19 +148,44 @@ async def update_and_get_progress(job_id: JobId) -> JobProgressResponse:
 
 @router.get("/status")
 async def get_status(
-    user = Depends(current_active_user),
+    user: UserRead = Depends(current_active_user),
     page: int = 1,
-    page_size: int = 10
+    page_size: int = 10,
+    status: STATUS | None = None
 ) -> JobProgressResponses:
-    """Get progress of all tasks recorded in the database with pagination."""
+    """Get progress of all tasks recorded in the database with pagination and filtering.
+
+    Parameters
+    ----------
+    user : UserRead
+        The current active user.
+    page : int
+        Page number (1-indexed).
+    page_size : int
+        Number of items per page.
+    status : STATUS | None
+        Filter by job status (submitted, running, completed, errored, invalid, timeout, unknown).
+
+    Returns
+    -------
+    JobProgressResponses
+        Paginated job progress responses with metadata.
+    """
 
     if page < 1 or page_size < 1:
         raise HTTPException(status_code=400, detail="Page and page_size must be greater than 0.")
 
+    # Get all jobs and apply filters
     job_records = list(await get_all())
+
+    # Apply status filter
+    if status:
+        job_records = [job for job in job_records if job.status == status]
```

Comment:
this one would better go to the db query, but I'll make the change later when I propagate down the `limit` etc

Reply:
you can merge this in the meantime tho